include/battle/battle_context.h

```diff
@@ -0,0 +1,294 @@
+    int battlerIdAbility;
+    int battlerIdMagicCoat;
+    int addlEffectType;             // differing from pokeheartgold here for accuracy
+    int addlEffectParam;
+    int addlEffectFlags;
+    int battlerIdAddlEffect;
+    int backupBattlerId;            // not sure what else to name this tbh
+    
+    int gainedExp;
+    int sharedExp;
+    u32 monsGainingExp[2];
+
+    int scriptNarcId;
+    int scriptFileId;
+    int scriptCursor;
+    int scriptStackPointer;
+    int scriptStackNarcId[BATTLE_SCRIPT_STACK_MAX];
+    int scriptStackFileId[BATTLE_SCRIPT_STACK_MAX];
+    int scriptStackCursor[BATTLE_SCRIPT_STACK_MAX];
+
+    int turnOrderCounter;
+    int waitCounter;
+
+    UnkStruct_ov16_0225C300 msgBuffer;      // generic holding buffer for pre-processed message
+    int msgBattlerIdTemp;
+    int msgBattlerIdAttacker;
+    int msgBattlerIdDefender;
+    int msgMoveNumTemp;
+    int msgItemNumTemp;
+    int msgAbilityNumTemp;
+    int msgTemp;
+    int calcTemp;
+    int scriptTemp;
+
+    u32 battlerStatusFlags[MAX_BATTLERS];
+    u32 payDayCount;
+    int totalTurns;
+    int totalFainted[MAX_BATTLERS];
+    int totalDamage[MAX_BATTLERS];
+    int meFirstTurnOrder;
+
+    UnkStruct_ov16_0224B7CC *unk_178;       // task control stuff
+    void *unk_17C;
+    
+    u32 fieldConditionsMask;
+    FieldConditions fieldConditions;
+
+    u32 sideConditionsMask[MAX_BATTLE_SIDES];
+    SideConditions sideConditions[MAX_BATTLE_SIDES];
+    
+    TurnFlags turnFlags[MAX_BATTLERS];
+    SelfTurnFlags selfTurnFlags[MAX_BATTLERS];
+    MoveFailFlags moveFailFlags[MAX_BATTLERS];
+
+    UnkStruct_ov14_0221FC20 aiData;         // has the move and item data tables
+    u32 *aiScriptTemp;
+    u32 aiScriptCursor;
+
+    u32 battleStatusMask;
+    u32 battleStatusMask2;
+    
+    int damage;
+    int hitDamage;
+    int criticalBoosts;
+    int criticalMulti;
+    int movePower;
+    int powerMulti;
+    int hpCalcTemp;
+    int moveType;
+    int moveEffectChance;
+    int prizeMoneyMulti;
+    u32 moveStatusFlags;
+    u32 addlEffectDirectFlags;
+    u32 addlEffectIndirectFlags;
+    u32 addlEffectAbilityFlags;
+    u8 multiHitCounter;
+    u8 multiHitNumHits;
+    u8 battlerIdCounter;
+    u8 beatUpCounter;
+
+    BOOL multiHitLoop;
+    u32 multiHitCheckFlags;
+    u32 multiHitAccuracyCheck;
+
+    u32 clearVolatileStatus[MAX_BATTLERS];
+
+    u8 selectedMonIndex[MAX_BATTLERS];
+    u8 switchSelectedMonIndex[MAX_BATTLERS];
+    u8 aiSwitchSelectedMonIndex[MAX_BATTLERS];
+    
+    u32 battlerActions[MAX_BATTLERS][MAX_BATTLE_ACTIONS];
+    u8 battlerActionOrder[MAX_BATTLERS];
+    u8 monSpeedOrder[MAX_BATTLERS];
+    u32 monSpeedValues[MAX_BATTLERS];
+    
+    u8 ioQueue[MAX_LINK_BATTLERS][MAX_BATTLERS][BATTLE_IO_QUEUE_SIZE];
+    u8 ioBuffer[MAX_BATTLERS][BATTLE_IO_BUFFER_SIZE];
+    int battleScript[BATTLE_SCRIPT_SIZE_MAX];
+
+    UnkStruct_ov16_02252060 battleMons[MAX_BATTLERS];
+
+    u32 moveNumTemp;
+    u32 moveNumCurr;
```

Comment:
I'm not gonna list this everywhere, but the most used short for current in this repo is `cur`, not `curr`.